### PR TITLE
Updated DNS entries so all subclasses of DNSRecord use to_string for __repr__

### PR DESCRIPTION
**Pull request by jhogg**
Cleaned up output of cache dumps with:

- All records based on DNSRecord now properly use to_string in __repr__, some were only dumping the answer without the question (inconsistent).

before:
 record[txt,in-unique,AD-R02-3._netaudio-cmc._udp.local.]=4500/3177,b'\x13id=000'...,
 record[txt,in-unique,04@AD-R02-3._netaudio-chan._udp.local.]=4500/4377,b'\ttxtver'...,
172.30.11.164,
 record[txt,in-unique,Iyo1616M-145b50._netaudio-arc._udp.local.]=4500/3177,b'\x10arcp_v'...,
 record[txt,in-unique,Iyo1616M-145b50._netaudio-dbc._udp.local.]=4500/4377,b'\x00',

after:
 record[txt,in-unique,AD-R02-3._netaudio-cmc._udp.local.]=4500/3177,b'\x13id=000'...,
 record[txt,in-unique,04@AD-R02-3._netaudio-chan._udp.local.]=4500/4377,b'\ttxtver'...,
 record[a,in-unique,Iyo1616M-145b50.local.]=120/88,172.30.11.164,
 record[txt,in-unique,Iyo1616M-145b50._netaudio-arc._udp.local.]=4500/3177,b'\x10arcp_v'...,
 record[txt,in-unique,Iyo1616M-145b50._netaudio-dbc._udp.local.]=4500/4377,b'\x00',

- Cleaned up output of ttl remaining to be whole seconds only (prior included 8 digits of accuracy)
- Cleaned up format to cleanly separate [question]=ttl,answer

### CODE REVIEW

File: zeroconf/__init__.py [hunk 1]
```diff
@@ -420,7 +420,7 @@ def entry_to_string(self, hdr: str, other: Optional[Union[bytes, str]]) -> str:
             result += ","
         result += self.name
         if other is not None:
-            result += ",%s]" % cast(Any, other)
+            result += "]=%s" % cast(Any, other)
```
Comment (jstasiak) [inline zeroconf/__init__.py]: Either this belongs in a separate pull request or the name of this one should be changed to "clean up string representations" or something – now it only says it's about using `to_string` in `__repr__` and this is clearly something extra.

File: zeroconf/__init__.py [hunk 2]
```diff
@@ -509,7 +509,7 @@ def write(self, out: 'DNSOutgoing') -> None:
 
     def to_string(self, other: Union[bytes, str]) -> str:
         """String representation with additional information"""
-        arg = "%s/%s,%s" % (self.ttl, self.get_remaining_ttl(current_time_millis()), cast(Any, other))
+        arg = "%s/%s,%s" % (self.ttl, int(self.get_remaining_ttl(current_time_millis())), cast(Any, other))
```
Comment (jstasiak) [inline zeroconf/__init__.py]: See my comment above. Either a separate pull request or this one should be renamed.

**Comment by jstasiak** [comment]
@jhogg I'm happy to merge the "__repr__ uses to_string" part of your patch manually, let me know if this is ok.

**Comment by jhogg** [comment]
@jstasiak Perfectly ok.  I intentionally split the functional (TTL issue) and non-functional (logging), and I get the request for more detailed commit, but I'm on another piece right now and would probably be next week before I split this patch on __repr__ vs "]=%s".  I did call out the 2 cleanups that were part of the __repr__ in the pull request notes (but not commit).

 I'm usually on a ticketing system, so just a ticket ref, have to remind my self otherwise on these.

If you want it later: These were cleaned up because I am expanding the logging to be able to watch cache changes (add,remove,update), ttl, answers and service browser interaction on queries.  This is all part from tracking down the TTL issue.

**Comment by jstasiak** [comment]
ok, I'll actually just split this into three commits and put your pull request notes as commit messagess in the right form if that's all right, as I have a moment available.

**Comment by jstasiak** [comment]
Released as part of 0.24.4 just now.
